### PR TITLE
Allow proxies for nvidia downloads via environment variables

### LINKED ISSUE #10: Cannot install through proxy - need to configure download url
I failed to install cudatoolkit-dev behind a corporate firewall (although with a proxy for conda), and only got a cryptic error:
```
ERROR conda.core.link:_execute(568): An error occurred while installing package 'çonda-forge::cudatoolkit-dev-10.0-1'.
LinkError: post-link script failed for package conda-forge::cudatoolkit-dev-10.0-1
running your command again with `-v` will provide additional information
location of failed script: /opt/conda/bin/.cudatoolkit-dev-post-link.sh
==> script messages <==
<None>

Attempting to roll back
```

What's not obvious at first is that the `cudatoolkit-dev-post-install.py` script tries to download files from a given url, which of course is blocked by our firewall:
https://github.com/conda-forge/cudatoolkit-dev-feedstock/blob/3bae06d62411106c41e1c1cf20897a5502dbbb66/recipe/cudatoolkit-dev-post-install.py#L316

What I'd need is to modify this URL so I can point it at a proxy (that I can - with some effort - get set up to circumvent the firewall). Will try to prepare a PR shortly.

### PR DESCRIPTION
**Pull request by h-vetinari**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes #10 

Happy to bikeshed over environment variable names. Unfortunately I can't avoid adding one each for developer.nvidia.com resp. developer.download.nvidia.com, because the files and md5s actually lie on separate urls.

### CODE REVIEW

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
@conda-forge/cudatoolkit-dev 
I managed that the builds complete with a rerendered recipe (on azure), but I cannot figure out the test errors on linux. I tried to build the recipe with the state of this PR on a Ubuntu 18.04 machine and it ran through without errors.

**Comment by andersy005** [comment]
Thank you for the PR, @h-vetinari! 

> I tried to build the recipe with the state of this PR on a Ubuntu 18.04 machine and it ran through without errors.

I've seen similar behavior in the past, and I don't know what causes it. I will try to re-run the CI builds and see what happens.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,10 +12,12 @@ source:
 
 build:
   script: python build.py {{ version }} {{ version_build }} {{ driver_version }}
-  number: 1
+  number: 2
   skip: True  # [win]
 
 requirements:
+  build:
+    - python >=3.6 [osx]  # osx only has python 2 by default
```
Comment (scopatz) [inline recipe/meta.yaml]: Selectors need to be behind a comment, ie maye this should be:

```yaml
# osx only has python 2 by default
- python >=3.6  # [osx]
```

Comment (h-vetinari) [reply recipe/meta.yaml]: Ah yes, silly oversight. I had swapped things around from this after the linter complained.

**Comment by scopatz** [comment]
Also, linux builds seems to be missing files now: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=45047

**Comment by h-vetinari** [comment]
It's the same error that showed up on mac before I added python to the build setup. Guess we need the build dep for linux too.

**Comment by h-vetinari** [comment]
So, now we're back to the `ClobberWarning` (that didn't leave by either specifying python or tqdm as a test dep), and the invisible test errors.

**Comment by scopatz** [comment]
Yeah I don't understand that one

**Comment by h-vetinari** [comment]
@andersy005 @xmnlab 
Any ideas what I can do here to fix the tests? @andersy005 said he already had similar issues before, but so far they don't seem to disappear just from restarting the CI...

**Comment by h-vetinari** [comment]
@conda-forge/core
Could anyone help debug this? Or at least, how to get an error log from that failed test?

**Comment by scopatz** [comment]
I think you can cat / echo a log file to the screen if you need it.

**Comment by h-vetinari** [comment]
The only thing I can find from googling about these ClobberWarnings is this [comment](https://github.com/conda-forge/configparser-feedstock/issues/19#issuecomment-464964277) from @jakirkham, where it was necessary to add a package (here tdqm?) to both host and run dependencies.

I tried that with the last few commits but the warnings (and failure) remains the same.

I'm not entirely clear what's happening with that clobber-stuff. Especially since it's complaining that it's clobbering an existing path, but then shows `source path: None` in the Warning...

File: recipe/meta.yaml [hunk 2]
```diff
@@ -12,12 +12,20 @@ source:
...
   skip: True  # [win]
 
 requirements:
+  build:
+    # default on linux/osx is python2
+    - python >=3.6
+  host:
+    - python >=3.6
+    - tqdm
   run:
```
Comment (andersy005) [inline recipe/meta.yaml]: As pointed out in #14 and  #12 ,  conda isn't really needed at runtime. So, it makes sense to drop this section

**Review by andersy005 — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (h-vetinari) [reply recipe/meta.yaml]: I tried removing it in https://github.com/conda-forge/cudatoolkit-dev-feedstock/pull/11/commits/24b834a65a2d91f8b77e56ead32d83cf1eca2314, but this caused a `CondaMultiError`. I haven't followed the rest of the changes in #14 though.

Comment (andersy005) [reply recipe/meta.yaml]: When you get time, could you give 
```yaml
requirements:
  host:
    - python >=3
    - conda
```
a try?

**Comment by h-vetinari** [comment]
@andersy005 
The build failed again, but curiously, I can't see a log, only:
```Run docker build failed```

**Comment by andersy005** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
Interestingly, the osx-build for https://github.com/conda-forge/cudatoolkit-dev-feedstock/pull/11/commits/aec76739a7ff7e222be098c09c0b4367700e4bd2 failed as well...

**Comment by h-vetinari** [comment]
Adding conda to host (I'm guessing without the other changes from #14) means we're back to the `CondaMultiError` in the linux build that I mentioned above. 

Maybe it's better to just focus on #14 for the time being?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -12,12 +12,13 @@ source:
...
-  number: 1
+  number: 2
   skip: True  # [win]
 
 requirements:
-  run:
-    - conda >=4.5.8
+  host:
+    - python >=3
+    - conda
```
Comment (andersy005) [inline recipe/meta.yaml]:
```suggestion
  test:
    requires:
      - conda
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -12,13 +12,16 @@ source:
...
-    - conda >=4.5.8
-
+  host:
+    - python >=3
+    - conda
+  test:
+    requires:
+      - conda
 test:
   commands:
```
Comment (andersy005) [inline recipe/meta.yaml:25]: I meant to put it right here instead

**Comment by andersy005** [comment]
@h-vetinari, I am going to try debugging the linux issue outside of the container on a linux machine and will get back to you....

**Comment by h-vetinari** [comment]
> @andersy005: @h-vetinari, I am going to try debugging the linux issue outside of the container on a linux machine and will get back to you....

Thanks! I tried to do the same a [while back](https://github.com/conda-forge/cudatoolkit-dev-feedstock/pull/11#issuecomment-505865555), and could build on Ubuntu 18.04 without a problem.

**Comment by andersy005** [comment]
> could build on Ubuntu 18.04 without a problem.

The CI is using `vmImage: ubuntu-16.04`, and I am not sure why this might be the culprit...

**Comment by h-vetinari** [comment]
> @andersy005: The CI is using vmImage: ubuntu-16.04, and not I am not sure why this might be the culprit...

Yeah, I'm pretty sure the difference is somewhere else. Also didn't have ClobberWarnings, etc. at the time.

**Comment by andersy005** [comment]
I was wrong about the VM being the issue.. The VM is running ubuntu-16-04, however, everything ends up running in a container with centos I think.

**Comment by andersy005** [comment]
@h-vetinari,

I was able to build the package on both `"Debian GNU/Linux 9 (stretch)"` and `"Ubuntu 18.04.2 LTS"`. At this point, I don't really know why building the package on centos fails.

**Comment by andersy005** [comment]
I found out that cudatoolkit installer is hard-coded to use perl from `/usr/bin/perl`. The current centos image doesn't come with perl. The available installation is in `/opt/conda/bin/perl`

```bash
/opt/conda/pkgs/cudatoolkit-dev/cuda_10.0.130_410.48_linux: ./cuda-installer.pl: /usr/bin/perl: bad interpreter: No such file or directory
```
```
[conda@7cf862609754 ~]$ which perl
/opt/conda/bin/perl
```

**Comment by andersy005** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
> @andersy005: I found out that cudatoolkit installer is hard-coded to use perl from `/usr/bin/perl`.

Thanks for the investigation! This can be easily added with `yum_requirements.txt`. Let's see if this works. :)

**Comment by h-vetinari** [comment]
OK, it didn't work. However, I can't see the yum install in the build log, so not sure what's happening there.

EDIT: the recipe needed to be rerendered, because the azure `build_steps.sh` gets generated based on (i.a.) the content of `yum_requirements.txt`

**Comment by h-vetinari** [comment]
@andersy005 
Success! :)

This now builds successfully, using the old tests (and my fix for the chmod issue). If you want to copy some of your other changes from #14, feel free.

**Comment by h-vetinari** [comment]
I rebased and cleaned up the history a bit, so that there's not so many garbage commits floating around (if you don't squash-merge).

**Review by andersy005 — APPROVED** [review]

**Comment by andersy005** [comment]
@h-vetinari, thank you for the PR and the additional fixes. I am merging this soon if there's no objection

**Comment by h-vetinari** [comment]
> @andersy005: @h-vetinari, thank you for the PR and the additional fixes. I am merging this soon if there's no objection

No problem! It was a team effort, because without your investigation about the hardlink to `/usr/bin/perl` I wouldn't have known what to do... ;-)

I rebased once more to correct a wrong commit message and add a comment to the yum-requirement.

**Comment by h-vetinari** [comment]
This should now also close #12 and therefore supersede #14. @jorisvandenbossche

**Comment by andersy005** [comment]
Closes #14

**Pull request merged**